### PR TITLE
Calculator in Javascript

### PR DESCRIPTION
**Pull request by burgosLeo96**
The folder JSCalculator has all the files developed for the homework: Calculator using javascript. It also uses the topic viewed in today's lesson.

### CODE REVIEW

File: bogota-july-2018/JSCalculator/assets/js/script.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+
+var operations = {
+	"+": (num1, num2) => num1 + num2,
+	"-": (num1, num2) => num1 - num2,
+	"*": (num1, num2) => num1 * num2,
+	"/": (num1, num2) => num2 !== 0 ? num1/num2 : "Error!, use Clear button",
```
Comment (globantPipo) [inline bogota-july-2018/JSCalculator/assets/js/script.js]: This error is not clear, the user needs to know the div0 issue

File: bogota-july-2018/JSCalculator/assets/js/script.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	"":(num1, num2) => num2
```
Comment (globantPipo) [inline bogota-july-2018/JSCalculator/assets/js/script.js]: Why do we need this function ?

File: bogota-july-2018/JSCalculator/assets/js/script.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+var keyCodes = {
+	"106": "*",
+	"107": "+",
+	"109": "-",
+	"111": "/"
+}
+
+var res = 0;
```
Comment (globantPipo) [inline bogota-july-2018/JSCalculator/assets/js/script.js]: Var naming is important, `result` is an explicit name that helps other devs to understand your code.

File: bogota-july-2018/JSCalculator/assets/js/script.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+var first = true;
+var equal = false;
+var opc = "";
+var num2 = 0;
+
+var input = document.querySelector("#input-numbers");
+var output = document.querySelector("output");
```
Comment (globantPipo) [inline bogota-july-2018/JSCalculator/assets/js/script.js]: Is NOT  a good idea to have this kind of selectors, if some other dev is NOT aware of this code, he can add another `output` breaking your code, please use a class.

File: bogota-july-2018/JSCalculator/assets/js/script.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+var options  =document.querySelectorAll(".operator");
+var equals = document.querySelector("#equals");
+var clear = document.querySelector("#clear");
+
+for(var i = 0; i < options.length; i++){
```
Comment (globantPipo) [inline bogota-july-2018/JSCalculator/assets/js/script.js]: Usually talking about performance is better to use a `forEach` statement ( i know maybe we didnt say that)

File: bogota-july-2018/JSCalculator/assets/js/script.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+	options[i].addEventListener("click", function(){
```
Comment (globantPipo) [inline bogota-july-2018/JSCalculator/assets/js/script.js]: Adding a LOT of evenListener also is not a good idea, here is a doc talking about event delegation that maybe can help u 
https://davidwalsh.name/event-delegate

File: bogota-july-2018/JSCalculator/assets/js/script.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+		}
+	
+	input.focus();
+	input.select();
+	input.value = "";
+	output.textContent = res;
+	});
+}
+
+equals.addEventListener("click", function(){
```
Comment (globantPipo) [inline bogota-july-2018/JSCalculator/assets/js/script.js]: Keep in mind that anonymus functions usually are harder to debug in larger codes.

File: bogota-july-2018/JSCalculator/assets/js/script.js (hunk 6 above)
Comment (globantPipo) [reply bogota-july-2018/JSCalculator/assets/js/script.js]: i know we didn't talk about that :)

**Comment by Sebas-Alarcon-WebUI** [comment]
Check comments done by Felipe

**Pull request closed without merging**